packages/admin-portal/src/components/ui/approval-modal/decline-revision-modal.tsx

```diff
@@ -53,13 +53,15 @@ export const DeclineRevisionModal: React.FunctionComponent<DeclineRevisionModalP
   const session = useReapitConnect(reapitConnectBrowserSession)
   const { email, name } = session?.connectSession?.loginIdentity || {}
 
-  const { declineFormState } = revisionDetail
+  // form state
+  const formState = useSelector(selectAppRevisionFormState)
 
-  const { appId, id: appRevisionId } = revisionDetail?.revisionDetailData?.data || {}
+  // extract
+  const { appId, id: appRevisionId } = (revisionDetail?.data?.data || {}) as AppRevisionModel
   const [rejectionReason, setRejectionReason] = React.useState('')
 
-  const isLoading = declineFormState === 'SUBMITTING'
-  const isSuccessed = declineFormState === 'SUCCESS'
+  const isLoading = formState === 'SUBMITTING'
```

Comment:
could you use isLoading from reducer instead of check formState

Reply:
Same as above